src/core/vanilla.ts

```diff
@@ -336,9 +336,7 @@ export const delAtom = (
   if (mounted) {
     const [dependents] = mounted
     dependents.delete(useId)
-    if (!dependents.size) {
-      unmountAtom(state, deletingAtom)
-    }
```

Comment:
@Aslemammad The intuition behind this is when this deletingAtom has other dependents, we shouldn't unmount it. Could you create some other test cases (I can think at least two cases (1) two components use an atom directly; (2) one component use it, another component use it indirectly), and see this change fails.